### ISSUE TITLE
No client-side tests but 2 fail

### ISSUE THREAD

**Issue by rhettlivingston** [DispatchMe/meteor-mocha-phantomjs#23]
I'm getting a bunch of typeErrors and 2 client side fails when using `meteor test --once --driver-package dispatch:mocha-phantomjs` but I have no client-side tests. Also, the command never terminates. 

`meteor test --once --driver-package practicalmeteor:mocha` works fine so the problem seems to be isolated to mocha-phantomjs.

It is apparently something unique to my app triggering it because I ran the same test command on the todos app with the same meteor and package versions shown below and didn't encounter this problem, but I'm at a loss as to what or where to look.

Possibly pertinent versions include:
- METEOR@1.4-beta.1
- dispatch:mocha-phantomjs@0.1.6
- dispatch:phantomjs-tests@0.0.5
- practicalmeteor:chai@2.1.0_1
- practicalmeteor:loglevel@1.2.0_2
- practicalmeteor:mocha@2.4.5_4
- practicalmeteor:mocha-core@1.0.0
- practicalmeteor:sinon@1.14.1_2

``` bash
I20160624-20:47:10.546(-4)?                   
I20160624-20:47:10.580(-4)? --------------------------------
I20160624-20:47:10.582(-4)? ----- RUNNING SERVER TESTS -----
I20160624-20:47:10.582(-4)? --------------------------------
I20160624-20:47:10.582(-4)? 
I20160624-20:47:10.582(-4)? 
I20160624-20:47:10.582(-4)? 
I20160624-20:47:10.582(-4)?   groups
I20160624-20:47:10.582(-4)?     methods
I20160624-20:47:10.583(-4)?       insert
I20160624-20:47:10.583(-4)?         positive tests
=> Started your app.

=> App running at: http://localhost:3000/
I20160624-20:47:10.786(-4)? 
I20160624-20:47:10.787(-4)?           ✓ creates a good group (111ms)
I20160624-20:47:10.855(-4)? 
... <cut out a bunch of similar stuff here>
I20160624-20:47:12.481(-4)? 
I20160624-20:47:12.482(-4)?           ✓ denies removal of last administrator (43ms)
I20160624-20:47:12.482(-4)? 
I20160624-20:47:12.482(-4)? 
I20160624-20:47:12.483(-4)?   38 passing (2s)
I20160624-20:47:12.483(-4)?   1 pending
I20160624-20:47:12.483(-4)? 
I20160624-20:47:12.483(-4)? 
I20160624-20:47:12.483(-4)? --------------------------------
I20160624-20:47:12.483(-4)? ----- RUNNING CLIENT TESTS -----
I20160624-20:47:12.483(-4)? --------------------------------
I20160624-20:47:12.483(-4)? 
I20160624-20:47:12.484(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.484(-4)? 
I20160624-20:47:12.484(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.485(-4)?     http://localhost:3000/packages/es5-shim.js?hash=e4d68c9ebbcdbd230d16f4bd933ee8dbf2978a22: 17
I20160624-20:47:12.486(-4)? 
I20160624-20:47:12.487(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.488(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.488(-4)? 
I20160624-20:47:12.488(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.488(-4)?     http://localhost:3000/packages/promise.js?hash=2d5b025794b9c9e54e5cf6bea71c642ef921a2ed: 17
I20160624-20:47:12.488(-4)? 
I20160624-20:47:12.488(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.489(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.489(-4)? 
I20160624-20:47:12.489(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.489(-4)?     http://localhost:3000/packages/ecmascript-runtime.js?hash=e4dd2b138377d446dc7dc3af21bb5bcd4b2b88a3: 17
I20160624-20:47:12.489(-4)? 
I20160624-20:47:12.489(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.489(-4)? TypeError: undefined is not an object (evaluating 'Package['ecmascript-runtime'].Symbol')
I20160624-20:47:12.489(-4)? 
I20160624-20:47:12.489(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.489(-4)?     http://localhost:3000/packages/babel-compiler.js?hash=aa1ba23957a2a38a24bad15d8eb7dda850adb69a: 7
I20160624-20:47:12.490(-4)? 
I20160624-20:47:12.490(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.490(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.490(-4)? 
I20160624-20:47:12.490(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.490(-4)?     http://localhost:3000/packages/babel-runtime.js?hash=543034092ecef720d1732d8f72ab489f1877696f: 17
I20160624-20:47:12.490(-4)? 
I20160624-20:47:12.490(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.496(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.496(-4)? 
I20160624-20:47:12.496(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.496(-4)?     http://localhost:3000/packages/random.js?hash=c4879267bde8bbc5ba66d23fa420cb239c44fb68: 18
I20160624-20:47:12.496(-4)? 
I20160624-20:47:12.496(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.497(-4)? TypeError: undefined is not an object (evaluating 'Package.random.Random')
I20160624-20:47:12.497(-4)? 
I20160624-20:47:12.497(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.497(-4)?     http://localhost:3000/packages/mongo-id.js?hash=172813d0ff26d0175642f30557d9980b3d852a17: 19
I20160624-20:47:12.497(-4)? 
I20160624-20:47:12.497(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.497(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.497(-4)? 
I20160624-20:47:12.497(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.497(-4)?     http://localhost:3000/packages/geojson-utils.js?hash=7214728d388f610508afa2a208b6117470822707: 17
I20160624-20:47:12.498(-4)? 
I20160624-20:47:12.498(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.498(-4)? TypeError: undefined is not an object (evaluating 'Package['mongo-id'].MongoID')
I20160624-20:47:12.498(-4)? 
I20160624-20:47:12.498(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.498(-4)?     http://localhost:3000/packages/minimongo.js?hash=88217d643bc16fdf3505c6d4b2b8f5ddc400c49a: 23
I20160624-20:47:12.498(-4)? 
I20160624-20:47:12.498(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.498(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.498(-4)? 
I20160624-20:47:12.498(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.499(-4)?     http://localhost:3000/packages/check.js?hash=bf95dee84c2be9008274d2385c637659760b9057: 17
I20160624-20:47:12.499(-4)? 
I20160624-20:47:12.499(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.499(-4)? TypeError: undefined is not an object (evaluating 'Package.random.Random')
I20160624-20:47:12.499(-4)? 
I20160624-20:47:12.499(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.499(-4)?     http://localhost:3000/packages/retry.js?hash=91c22616ae67cee99080f81fae9e6083bf7c5711: 18
I20160624-20:47:12.499(-4)? 
I20160624-20:47:12.499(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.499(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.500(-4)? 
I20160624-20:47:12.500(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.500(-4)?     http://localhost:3000/packages/ddp-common.js?hash=a99cb35fd21daff658b592895ec8e2da5ee4c434: 17
I20160624-20:47:12.500(-4)? 
I20160624-20:47:12.500(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.501(-4)? TypeError: undefined is not an object (evaluating 'Package['ecmascript-runtime'].Symbol')
I20160624-20:47:12.501(-4)? 
I20160624-20:47:12.501(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.501(-4)?     http://localhost:3000/packages/reload.js?hash=ad99c3e418c3fdc4fd1e08796369bded98b26b3d: 18
I20160624-20:47:12.501(-4)? 
I20160624-20:47:12.501(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.501(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.501(-4)? 
I20160624-20:47:12.502(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.502(-4)?     http://localhost:3000/packages/ddp-client.js?hash=27502404fad7fc072e57e8b0b6719f40d92709c7: 17
I20160624-20:47:12.502(-4)? 
I20160624-20:47:12.502(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.502(-4)? TypeError: undefined is not an object (evaluating 'Package['ddp-client'].DDP')
I20160624-20:47:12.502(-4)? 
I20160624-20:47:12.502(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.502(-4)?     http://localhost:3000/packages/ddp.js?hash=752b775e282ca59c21ca09b9cc61f5fffb9ea2a1: 7
I20160624-20:47:12.502(-4)? 
I20160624-20:47:12.502(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.502(-4)? TypeError: undefined is not an object (evaluating 'Package.minimongo.LocalCollection')
I20160624-20:47:12.503(-4)? 
I20160624-20:47:12.503(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.503(-4)?     http://localhost:3000/packages/allow-deny.js?hash=11ff6d1a797cb3d6608a96e8ab0bd82a4947ff49: 18
I20160624-20:47:12.503(-4)? 
I20160624-20:47:12.503(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.503(-4)? TypeError: undefined is not an object (evaluating 'Package['allow-deny'].AllowDeny')
I20160624-20:47:12.503(-4)? 
I20160624-20:47:12.503(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.503(-4)?     http://localhost:3000/packages/mongo.js?hash=4c6ab6b82bf423dbc0eb1e4fb8f736843b793ddd: 17
I20160624-20:47:12.503(-4)? 
I20160624-20:47:12.503(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.504(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.504(-4)? 
I20160624-20:47:12.504(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.504(-4)?     http://localhost:3000/packages/reactive-dict.js?hash=efe4d7acf43d54e77a8d1b428c7cc15f6e132898: 21
I20160624-20:47:12.504(-4)? 
I20160624-20:47:12.504(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.504(-4)? TypeError: undefined is not an object (evaluating 'Package['reactive-dict'].ReactiveDict')
I20160624-20:47:12.504(-4)? 
I20160624-20:47:12.504(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.504(-4)?     http://localhost:3000/packages/session.js?hash=9ac45190b217c0c6a6293b0769e5a83f9dc8c003: 18
I20160624-20:47:12.504(-4)? 
I20160624-20:47:12.505(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.505(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.505(-4)? 
I20160624-20:47:12.505(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.505(-4)?     http://localhost:3000/packages/jquery.js?hash=22a0055f59bd150c435c5aba34c7c59076b8bcd9: 17
I20160624-20:47:12.505(-4)? 
I20160624-20:47:12.505(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.505(-4)? TypeError: undefined is not an object (evaluating 'Package['reactive-dict'].ReactiveDict')
I20160624-20:47:12.505(-4)? 
I20160624-20:47:12.505(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.505(-4)?     http://localhost:3000/packages/kadira_flow-router.js?hash=09ea12875d3801955ee70797bf8e4a70feebc570: 20
I20160624-20:47:12.506(-4)? 
I20160624-20:47:12.506(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.506(-4)? TypeError: undefined is not an object (evaluating 'Package['mongo-id'].MongoID')
I20160624-20:47:12.506(-4)? 
I20160624-20:47:12.506(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.506(-4)?     http://localhost:3000/packages/observe-sequence.js?hash=550c39b36ab0e65194ea03cdc7ecbe99dcdd07f6: 19
I20160624-20:47:12.506(-4)? 
I20160624-20:47:12.506(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.506(-4)? TypeError: undefined is not an object (evaluating 'Package.jquery.$')
I20160624-20:47:12.506(-4)? 
I20160624-20:47:12.506(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.507(-4)?     http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea: 17
I20160624-20:47:12.507(-4)? 
I20160624-20:47:12.507(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.507(-4)? TypeError: undefined is not an object (evaluating 'Package.blaze.Blaze')
I20160624-20:47:12.507(-4)? 
I20160624-20:47:12.508(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.508(-4)?     http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5: 20
I20160624-20:47:12.508(-4)? 
I20160624-20:47:12.508(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.508(-4)? TypeError: undefined is not an object (evaluating 'Package.blaze.Blaze')
I20160624-20:47:12.508(-4)? 
I20160624-20:47:12.508(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.508(-4)?     http://localhost:3000/packages/templating.js?hash=f43427e80c4a8a93daf5ed094c8141a0ee19c804: 18
I20160624-20:47:12.508(-4)? 
I20160624-20:47:12.509(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.509(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.509(-4)? 
I20160624-20:47:12.509(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.509(-4)?     http://localhost:3000/packages/zimme_active-route.js?hash=eef5899b049cf07c5934903b338f3308a572cf27: 17
I20160624-20:47:12.509(-4)? 
I20160624-20:47:12.509(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.509(-4)? TypeError: undefined is not an object (evaluating 'Package.random.Random')
I20160624-20:47:12.510(-4)? 
I20160624-20:47:12.510(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.510(-4)?     http://localhost:3000/packages/localstorage.js?hash=0b3ae3298494da2289abab2377342a1054cfca0b: 17
I20160624-20:47:12.510(-4)? 
I20160624-20:47:12.510(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.510(-4)? TypeError: undefined is not an object (evaluating 'Package.random.Random')
I20160624-20:47:12.510(-4)? 
I20160624-20:47:12.510(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.510(-4)?     http://localhost:3000/packages/accounts-base.js?hash=3f6353a90da402397ef0b889e3812a40c0a71704: 20
I20160624-20:47:12.510(-4)? 
I20160624-20:47:12.510(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.510(-4)? TypeError: undefined is not an object (evaluating 'Package.random.Random')
I20160624-20:47:12.511(-4)? 
I20160624-20:47:12.511(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.511(-4)?     http://localhost:3000/packages/srp.js?hash=94d5df94b266ef008d0425c828f1715512e429ef: 17
I20160624-20:47:12.511(-4)? 
I20160624-20:47:12.511(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.511(-4)? TypeError: undefined is not an object (evaluating 'Package['accounts-base'].Accounts')
I20160624-20:47:12.511(-4)? 
I20160624-20:47:12.511(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.511(-4)?     http://localhost:3000/packages/accounts-password.js?hash=26c5712042d53cb92bed4571735603904d1ec8bd: 17
I20160624-20:47:12.511(-4)? 
I20160624-20:47:12.511(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.512(-4)? TypeError: undefined is not an object (evaluating 'Package['accounts-base'].Accounts')
I20160624-20:47:12.512(-4)? 
I20160624-20:47:12.512(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.512(-4)?     http://localhost:3000/packages/useraccounts_core.js?hash=7012276935817ce37e8eff40bf284912a2df8f53: 17
I20160624-20:47:12.512(-4)? 
I20160624-20:47:12.512(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.512(-4)? TypeError: undefined is not an object (evaluating 'Package.blaze.Blaze')
I20160624-20:47:12.512(-4)? 
I20160624-20:47:12.512(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.512(-4)?     http://localhost:3000/packages/kadira_blaze-layout.js?hash=493d22b2043f66d6b3694baa790b62a701aaec33: 17
I20160624-20:47:12.512(-4)? 
I20160624-20:47:12.513(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.513(-4)? TypeError: undefined is not an object (evaluating 'Package['ecmascript-runtime'].Symbol')
I20160624-20:47:12.513(-4)? 
I20160624-20:47:12.513(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.513(-4)?     http://localhost:3000/packages/softwarerero_accounts-t9n.js?hash=44df5d0a25a66b892314a1b84a5da25576e0fb50: 19
I20160624-20:47:12.513(-4)? 
I20160624-20:47:12.513(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.513(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.513(-4)? 
I20160624-20:47:12.513(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.513(-4)?     http://localhost:3000/packages/useraccounts_flow-routing.js?hash=b8cfe374b32e12a395cdac7ec35ff037d5c34e71: 17
I20160624-20:47:12.513(-4)? 
I20160624-20:47:12.514(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.514(-4)? TypeError: undefined is not an object (evaluating 'Package.templating.Template')
I20160624-20:47:12.514(-4)? 
I20160624-20:47:12.514(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.514(-4)?     http://localhost:3000/packages/useraccounts_unstyled.js?hash=547d44e879d73b9c4b3f01194060aa37e131aabd: 17
I20160624-20:47:12.514(-4)? 
I20160624-20:47:12.514(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.514(-4)? TypeError: undefined is not an object (evaluating 'Package.templating.Template')
I20160624-20:47:12.514(-4)? 
I20160624-20:47:12.514(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.514(-4)?     http://localhost:3000/packages/utilities_avatar.js?hash=eb514cb5e23bb2346c84b26ca139da6656b804ee: 17
I20160624-20:47:12.515(-4)? 
I20160624-20:47:12.515(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.515(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.515(-4)? 
I20160624-20:47:12.515(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.515(-4)?     http://localhost:3000/packages/aldeed_simple-schema.js?hash=a36531b7660faa5522e7fade0b3f4daf5b4b6641: 20
I20160624-20:47:12.515(-4)? 
I20160624-20:47:12.515(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.515(-4)? TypeError: undefined is not an object (evaluating 'Package['aldeed:simple-schema'].SimpleSchema')
I20160624-20:47:12.515(-4)? 
I20160624-20:47:12.515(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.516(-4)?     http://localhost:3000/packages/aldeed_collection2-core.js?hash=78aa4a4dfcd4f54f1b080164dd6ca648f9f16f75: 17
I20160624-20:47:12.516(-4)? 
I20160624-20:47:12.516(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.516(-4)? TypeError: undefined is not an object (evaluating 'Package['aldeed:collection2-core'].Collection2')
I20160624-20:47:12.516(-4)? 
I20160624-20:47:12.516(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.516(-4)?     http://localhost:3000/packages/aldeed_schema-index.js?hash=da21b578f26d8a36d658a3af94517e1e4328eddb: 17
I20160624-20:47:12.516(-4)? 
I20160624-20:47:12.516(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.517(-4)? TypeError: undefined is not an object (evaluating 'Package['aldeed:collection2-core'].Collection2')
I20160624-20:47:12.517(-4)? 
I20160624-20:47:12.517(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.517(-4)?     http://localhost:3000/packages/aldeed_schema-deny.js?hash=44955b65d156080d05a87e961fd692d65185e52b: 17
I20160624-20:47:12.517(-4)? 
I20160624-20:47:12.517(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.517(-4)? TypeError: undefined is not an object (evaluating 'Package['aldeed:collection2-core'].Collection2')
I20160624-20:47:12.517(-4)? 
I20160624-20:47:12.517(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.517(-4)?     http://localhost:3000/packages/aldeed_collection2.js?hash=5ec6fbe0b5398f1a92bcb85b39683807a681c0b1: 7
I20160624-20:47:12.517(-4)? 
I20160624-20:47:12.518(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.518(-4)? TypeError: undefined is not an object (evaluating 'Package.mongo.Mongo')
I20160624-20:47:12.518(-4)? 
I20160624-20:47:12.519(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.519(-4)?     http://localhost:3000/packages/dburles_collection-helpers.js?hash=cc6238423ea044218119084c04f6e10e824f2408: 18
I20160624-20:47:12.519(-4)? 
I20160624-20:47:12.519(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.519(-4)? TypeError: undefined is not an object (evaluating 'Package.mongo.Mongo')
I20160624-20:47:12.519(-4)? 
I20160624-20:47:12.519(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.519(-4)?     http://localhost:3000/packages/mquandalle_collection-mutations.js?hash=f5daf351a0602bcba4cbe86812c84ef1278e069f: 17
I20160624-20:47:12.519(-4)? 
I20160624-20:47:12.519(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.520(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.520(-4)? 
I20160624-20:47:12.520(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.520(-4)?     http://localhost:3000/packages/cottz_publish-relations.js?hash=8b3c3336ac314babceb6e40e7f6372f41cd690ad: 17
I20160624-20:47:12.520(-4)? 
I20160624-20:47:12.520(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.521(-4)? TypeError: undefined is not an object (evaluating 'Package.mongo.Mongo')
I20160624-20:47:12.521(-4)? 
I20160624-20:47:12.521(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.521(-4)?     http://localhost:3000/packages/matb33_collection-hooks.js?hash=b4fb31946b843c27a27119ed667cdfb53fd1d510: 17
I20160624-20:47:12.522(-4)? 
I20160624-20:47:12.522(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.522(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.522(-4)? 
I20160624-20:47:12.522(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.522(-4)?     http://localhost:3000/packages/arillo_flow-router-helpers.js?hash=ec9ac46c8e2f676aa461291920959a020f98eb93: 17
I20160624-20:47:12.523(-4)? 
I20160624-20:47:12.523(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.523(-4)? TypeError: undefined is not an object (evaluating 'Package.jquery.$')
I20160624-20:47:12.523(-4)? 
I20160624-20:47:12.523(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.523(-4)?     http://localhost:3000/packages/coursierprive_tether.js?hash=17c32fc80a9e55c0bd1394dd7c4b670bad3c79d0: 17
I20160624-20:47:12.523(-4)? 
I20160624-20:47:12.523(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.523(-4)? TypeError: undefined is not an object (evaluating 'Package.jquery.$')
I20160624-20:47:12.524(-4)? 
I20160624-20:47:12.524(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.524(-4)?     http://localhost:3000/packages/twbs_bootstrap.js?hash=f068e78f7c5fb7cfd5ffac7577060bad7e6c08f1: 17
I20160624-20:47:12.524(-4)? 
I20160624-20:47:12.524(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.524(-4)? TypeError: undefined is not an object (evaluating 'Package.session.Session')
I20160624-20:47:12.524(-4)? 
I20160624-20:47:12.524(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.524(-4)?     http://localhost:3000/packages/tap_i18n.js?hash=e6b02e8b2dd8d10b36659d5cba6333ae2d1c3be8: 22
I20160624-20:47:12.524(-4)? 
I20160624-20:47:12.525(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.525(-4)? TypeError: undefined is not an object (evaluating 'Package.blaze.Blaze')
I20160624-20:47:12.525(-4)? 
I20160624-20:47:12.525(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.525(-4)?     http://localhost:3000/packages/ui.js?hash=d0665bbf032532e2284f8843b30500babb10dd4d: 7
I20160624-20:47:12.525(-4)? 
I20160624-20:47:12.525(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.525(-4)? TypeError: undefined is not an object (evaluating 'Package.blaze.Blaze')
I20160624-20:47:12.525(-4)? 
I20160624-20:47:12.525(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.525(-4)?     http://localhost:3000/packages/raix_handlebar-helpers.js?hash=d4495494d785afb74c9d9e9d9ed76f895ef03f70: 17
I20160624-20:47:12.526(-4)? 
I20160624-20:47:12.526(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.526(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.526(-4)? 
I20160624-20:47:12.526(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.526(-4)?     http://localhost:3000/packages/mdg_validated-method.js?hash=97f6fd0879eece3a6df9b2b6ee43d0d96cbe6e74: 17
I20160624-20:47:12.526(-4)? 
I20160624-20:47:12.526(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.526(-4)? TypeError: undefined is not an object (evaluating 'Package.mongo.Mongo')
I20160624-20:47:12.526(-4)? 
I20160624-20:47:12.526(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.527(-4)?     http://localhost:3000/packages/xolvio_cleaner.js?hash=24ad985e6756002606b4afe3dcecbdcded52b303: 17
I20160624-20:47:12.527(-4)? 
I20160624-20:47:12.527(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.527(-4)? TypeError: undefined is not an object (evaluating 'Package['aldeed:simple-schema'].SimpleSchema')
I20160624-20:47:12.527(-4)? 
I20160624-20:47:12.527(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.527(-4)?     http://localhost:3000/packages/rlivingston_simple-schema-mixin.js?hash=d7ac43f2b250a0e07b1e0a166ec06db0413bb0d3: 17
I20160624-20:47:12.527(-4)? 
I20160624-20:47:12.527(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.527(-4)? TypeError: undefined is not an object (evaluating 'Package['mdg:validated-method'].ValidatedMethod')
I20160624-20:47:12.527(-4)? 
I20160624-20:47:12.527(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.528(-4)?     http://localhost:3000/packages/rlivingston_simply-validated-method.js?hash=9a41df087500f36e6c1490dc6a7f8d7058dd23e2: 17
I20160624-20:47:12.528(-4)? 
I20160624-20:47:12.528(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.528(-4)? TypeError: undefined is not an object (evaluating 'Package.templating.Template')
I20160624-20:47:12.528(-4)? 
I20160624-20:47:12.528(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.528(-4)?     http://localhost:3000/packages/meteortoys_toykit.js?hash=ce7befbeda52596e5dbc7bd1f0c1137ae7ce88c9: 17
I20160624-20:47:12.528(-4)? 
I20160624-20:47:12.528(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.528(-4)? TypeError: undefined is not an object (evaluating 'Package.templating.Template')
I20160624-20:47:12.528(-4)? 
I20160624-20:47:12.529(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.529(-4)?     http://localhost:3000/packages/msavin_mongol.js?hash=cf20c0f46ab44ddf9413cf9078a324ff0104cf8c: 17
I20160624-20:47:12.529(-4)? 
I20160624-20:47:12.529(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.529(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.529(-4)? 
I20160624-20:47:12.529(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.529(-4)?     http://localhost:3000/packages/practicalmeteor_mocha-core.js?hash=7005f838fdbcbc39c2b96bd08f49d548b21d9440: 17
I20160624-20:47:12.529(-4)? 
I20160624-20:47:12.529(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.529(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.530(-4)? 
I20160624-20:47:12.530(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.530(-4)?     http://localhost:3000/packages/dispatch_mocha-phantomjs.js?hash=0a4b98531e0c8e6d5eb7acb96f8e3202ac5e69ff: 17
I20160624-20:47:12.530(-4)? 
I20160624-20:47:12.530(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.530(-4)? TypeError: undefined is not an object (evaluating 'Package.jquery.$')
I20160624-20:47:12.530(-4)? 
I20160624-20:47:12.530(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.530(-4)?     http://localhost:3000/packages/tmeasday_test-reporter-helpers.js?hash=eff40c069227d810b2820a9b174f4fd737d2a3c8: 18
I20160624-20:47:12.530(-4)? 
I20160624-20:47:12.530(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.531(-4)? TypeError: undefined is not an object (evaluating 'Package.retry.Retry')
I20160624-20:47:12.531(-4)? 
I20160624-20:47:12.531(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.531(-4)?     http://localhost:3000/packages/autoupdate.js?hash=96236e059901937f5f6e64a2cc27f899a294ad73: 19
I20160624-20:47:12.531(-4)? 
I20160624-20:47:12.531(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.531(-4)? TypeError: undefined is not an object (evaluating 'Package['ecmascript-runtime'].Symbol')
I20160624-20:47:12.531(-4)? 
I20160624-20:47:12.531(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.531(-4)?     http://localhost:3000/packages/practicalmeteor_chai.js?hash=6b28977253ebb6bb7f7f7a1424069e030aa224e2: 7
I20160624-20:47:12.532(-4)? 
I20160624-20:47:12.532(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.532(-4)? TypeError: undefined is not an object (evaluating 'Package['practicalmeteor:chai'].chai')
I20160624-20:47:12.532(-4)? 
I20160624-20:47:12.532(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.532(-4)?     http://localhost:3000/packages/practicalmeteor_loglevel.js?hash=96cd6887e3b954341d356b2fb1373d3149d292c4: 17
I20160624-20:47:12.532(-4)? 
I20160624-20:47:12.532(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.532(-4)? TypeError: undefined is not an object (evaluating 'Package['practicalmeteor:chai'].chai')
I20160624-20:47:12.532(-4)? 
I20160624-20:47:12.532(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.533(-4)?     http://localhost:3000/packages/practicalmeteor_sinon.js?hash=d51f09b72f5e645e87a6101ca90a5e06f108616d: 7
I20160624-20:47:12.533(-4)? 
I20160624-20:47:12.533(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.533(-4)? TypeError: undefined is not an object (evaluating 'Package.reload.Reload')
I20160624-20:47:12.533(-4)? 
I20160624-20:47:12.533(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.533(-4)?     http://localhost:3000/packages/practicalmeteor_mocha.js?hash=48311d8d578694d27de704142d6093ebe2dc3fbe: 17
I20160624-20:47:12.533(-4)? 
I20160624-20:47:12.533(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.533(-4)? TypeError: undefined is not an object (evaluating 'Package.modules.meteorInstall')
I20160624-20:47:12.533(-4)? 
I20160624-20:47:12.533(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.534(-4)?     http://localhost:3000/packages/webapp.js?hash=b74e2292d6087dc6f77d86b7e71ac0d672b0fbe8: 18
I20160624-20:47:12.534(-4)? 
I20160624-20:47:12.534(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.534(-4)? TypeError: undefined is not an object (evaluating 'Package['ddp-client'].DDP')
I20160624-20:47:12.534(-4)? 
I20160624-20:47:12.534(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.534(-4)?     http://localhost:3000/packages/livedata.js?hash=e7fb2e783bc81b00b914c2e27966b7b6d3d081b6: 7
I20160624-20:47:12.534(-4)? 
I20160624-20:47:12.534(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.534(-4)? TypeError: undefined is not an object (evaluating 'Package.mongo.Mongo')
I20160624-20:47:12.534(-4)? 
I20160624-20:47:12.535(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.535(-4)?     http://localhost:3000/packages/global-imports.js?hash=b715e8c44f45805378b9eb6cbddabe2321ed224d: 3
I20160624-20:47:12.535(-4)? 
I20160624-20:47:12.535(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.535(-4)? ReferenceError: Can't find variable: meteorInstall
I20160624-20:47:12.535(-4)? 
I20160624-20:47:12.535(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.535(-4)?     http://localhost:3000/app/app.js?hash=de46d23dc7f68e2bca9e609cc438b6482ccec789: 1
I20160624-20:47:12.535(-4)? 
I20160624-20:47:12.535(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.535(-4)? TypeError: undefined is not an object (evaluating 'Package['accounts-base'].Accounts')
I20160624-20:47:12.536(-4)? 
I20160624-20:47:12.536(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.536(-4)?     http://localhost:3000/packages/service-configuration.js?hash=331fe144dc8f69e5e44f23dd74b9de74ed80c6f3: 17
I20160624-20:47:12.536(-4)? 
I20160624-20:47:12.536(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.536(-4)? TypeError: undefined is not an object (evaluating 'Package.check.check')
I20160624-20:47:12.536(-4)? 
I20160624-20:47:12.536(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.536(-4)?     http://localhost:3000/packages/mdg_validation-error.js?hash=51a46aaba26a0e523bb856b942a6c42de1d3e66e: 17
I20160624-20:47:12.536(-4)? 
I20160624-20:47:12.536(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.536(-4)? Error: Can't find test driver package: dispatch:mocha-phantomjs
I20160624-20:47:12.537(-4)? 
I20160624-20:47:12.537(-4)?   phantomjs://code/phantomjsScript.js:15 in onError
I20160624-20:47:12.537(-4)?     http://localhost:3000/packages/meteor.js?hash=ae8b8affa9680bf9720bd8f7fa112f13a62f71c3: 1019
I20160624-20:47:12.537(-4)? 
I20160624-20:47:12.537(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.537(-4)?     http://localhost:3000/packages/meteor.js?hash=ae8b8affa9680bf9720bd8f7fa112f13a62f71c3: 809
I20160624-20:47:12.537(-4)? 
I20160624-20:47:12.537(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:12.537(-4)?     http://localhost:3000/packages/meteor.js?hash=ae8b8affa9680bf9720bd8f7fa112f13a62f71c3: 821
I20160624-20:47:12.537(-4)? 
I20160624-20:47:12.537(-4)?   phantomjs://code/phantomjsScript.js:17
I20160624-20:47:40.611(-4)? All client and server tests finished!
I20160624-20:47:40.611(-4)? 
I20160624-20:47:40.615(-4)? --------------------------------
I20160624-20:47:40.615(-4)? SERVER FAILURES: 0
I20160624-20:47:40.615(-4)? CLIENT FAILURES: 2
I20160624-20:47:40.615(-4)? --------------------------------
```

**Comment by aldeed** [comment]
Try with dispatch:mocha-phantomjs@0.1.7. Probably won't fix this, but worth a try. Otherwise it might be related to https://github.com/practicalmeteor/meteor-mocha/issues/59

I won't be able to figure out anything more unless you can post a link to a simple app that reproduces the issue.

**Comment by rhettlivingston** [comment]
Thank you for your attention to the matter.

You are correct in that mocha-phantomjs@0.1.7 did not fix it. Because I can run the tests in the browser without a problem, I'm not chomping at the bit for a fix. If only effects my ability to implement travis (one of the checkmarks in my future).

I can wait to see whether practicalmeteor/meteor-mocha#59 fixes it.

If that fails us,,,
It's not super small, but [this version of my oob project](https://github.com/ofonebody/oob-meteor/tree/28af51beef9b1af1457b33aff8ee35deda5b8ee9) exhibits the error with Meteor 1.4-beta.7. Seeing it would involve the normal git clone, meteor npm install, meteor npm run test (after making sure your meteor symlink truly points to the meteor 1.4-beta.7 bundle).

**Comment by aldeed** [comment]
btw, if you will never have client tests, then you should just use https://github.com/DispatchMe/meteor-mocha instead

**Comment by abernix** [comment]
@rhettlivingston Can you post a Gist/PasteBin link to your `.meteor/packages` file to see what versions of various packages you have?

**Comment by rhettlivingston** [comment]
[Packages](https://github.com/ofonebody/oob-meteor/blob/28af51beef9b1af1457b33aff8ee35deda5b8ee9/.meteor/packages). 
[Versions](https://github.com/ofonebody/oob-meteor/blob/28af51beef9b1af1457b33aff8ee35deda5b8ee9/.meteor/versions). 
[This is a version of the project that is exhibiting the problem](https://github.com/ofonebody/oob-meteor/tree/28af51beef9b1af1457b33aff8ee35deda5b8ee9).

If you go to the latest version of that project, the problem is still there.

The problem does go away if I use meteor-mocha (or really, any other reporter) as aldeed suggested above. That is my current workaround but not a long-term solution as I will eventually need client-side unit testing.

I have seen meteor/todos exhibit a very similar dump when I went into a test file and created a reference error. It happened when I was debugging a tool for auto-generating import statements and an import was changed from something like `import { list } from './list';` to `import list from './list';`. Of course, that was fundamentally different in that the tests legitimately fail due to the reference errors. It was just similar in that it contained the same long chain of TypeErrors.

**Comment by abernix** [comment]
I am running into a similar problem, and it's quite bizarre.  On my project, the problem goes away if I remove `semantic:ui-css` OR `juliancwirko:s-alert`.  Both of those have a `jquery` dependency.

The problem also goes away (without any changes to my project at all) if I clone [DispatchMe/meteor-phantomjs-tests](https://github.com/DispatchMe/meteor-phantomjs-tests) into my project `packages` directory and change [the phantomjs NPM `depends`](https://github.com/DispatchMe/meteor-phantomjs-tests/blob/master/package.js#L11) from `phantomjs-prebuilt@2.1.7` to `phantomjs@1.9.20` (the 1.x `phantomjs` NPM was a different name) and also update the associated usage [here](https://github.com/DispatchMe/meteor-phantomjs-tests/blob/master/server.js#L1) to just `phantomjs`.

I actually have run into this problem in the past as well back with Velocity tests and if I recall, it boiled down to `Function.prototype.bind` being native in phantomjs2 whereas it was polyfilled in 1.x.  (or something, I can't find the issue but Sanjo/meteor-jasmine#237 comes close).  I remember explicitly specifying my `PHANTOMJS_BIN` to be a 1.0 release instead of 2.0 and it resolved it.

I'm going to dig around a bit more, but just thought I'd leave this.  Thanks for the information above – could be helpful.

**Comment by abernix** [comment]
@rhettlivingston btw, that https://github.com/ofonebody/oob-meteor/tree/28af51beef9b1af1457b33aff8ee35deda5b8ee9 is a perfect reproduction, thanks again.

Bit of progress!  It seems that there is one very likely omission from the errors you see in the client tests – I only found this by running PhantomJS with remote debugger enabled:

    SyntaxError: Unexpected token 'const'

...followed by all the errors that actually print to the log.

![screen shot 2016-07-15 at 8 14 00 pm](https://cloud.githubusercontent.com/assets/841294/16882627/b940597a-4ac8-11e6-9ab2-17266b1e7ed7.png)

Anyhow, I suspect that is the problem and it has something to do with ES6 not getting compiled somewhere and throwing off the rest of the compilation.  Not sure where yet! :wink:

**Comment by rhettlivingston** [comment]
You're welcome!

If there is a general weakness I've noted with the mocha related testing tools in Meteor it has been silent errors. In addition to finding the problem, hopefully someone will also consider why it was silenced and see if that can be corrected.

**Comment by abernix** [comment]
@rhettlivingston Ok, I solved your problem (and will likely solve mine soon thanks to PhantomJS remote debugging).

The `activitystrea.ms`NPM module you're using isn't meant to be used in the browser and doesn't transpile their source to ES5 code (since it's only expected to run on Node 4+).  Unfortunately, [it doesn't set a `browser` key in it's `package.json`](https://github.com/jasnell/activitystrea.ms/blob/master/package.json) indicating that it shouldn't be bundled for the browser, so it Meteor bundles the ES6 code into your browser bundle.  Meteor doesn't transpile it (because NPMs aren't meant to be re-transpiled) and drops it in your browser bundle.

Then, PhantomJS comes along and tries to read the module but doesn't support ES6, so it barfs with the above `SyntaxError`, which **for some reason is not passed to `phantom webpage`'s [`onError` in `dispatch:mocha-phantomjs`](https://github.com/DispatchMe/meteor-phantomjs-tests/blob/a659a87abf97fe433059fcadf54139032e4cc534/phantomjsScript.js#L10-L22)**. 🙈 

My solution?  I just went into `node_modules/activitystrea.ms/package.json` in your project directory and added:

    "browser": "./src/browser.js",

and then created an empty `node_modules/activitystrea.ms/src/browser.js` file.

Problem solved.  Obviously as soon as `activitystrea.ms` is updated, that change would get blown away.  I think it would be reasonable for `activitystrea.ms` to have a browser entry-point that does nothing (as I created), but that's up to the author.

I believe other test runners have no problem because they use actual browsers like Chrome which support (some) ES6 (like `const`).

**Comment by rhettlivingston** [comment]
@abernix Awesome! Thank you!

I suppose I should be thankful that PhantomJS uses an internal browser that lacks features. There may be some real browser where this would have failed.

I guess we're going to have to get used to such things as we join the NPM community. Our use cases are a bit of a mismatch with the larger community.

Before I close this, is there some change that could be made so that a problem like this is apparent to users without digging into the level that you did? Or is it the nature of the beast?

**Comment by abernix** [comment]
Let's leave it open for a bit.  I'm not super familiar with this PhantomJS stuff, but perhaps @aldeed will know a bit more about it as he has more experience.  I'm particular curious why the `SyntaxError` isn't passed through to Mocha and the logs.  I believe it might be a PhantomJS shortcoming, but I'm honestly not sure.

I just found the reason for my mocha tests failures and it's unrelated (and bizarre!), but it would have been super helpful if my `SyntaxError` was printed to the console.

**Comment by chmanie** [comment]
@abernix I'm having the same error (and I think it might be a SyntaxError as well), but even using remote debugging I can't see the SyntaxError (even here the first line is the TypeError).

Which version of phantomjs are you using and how does your debug process look like?

**Comment by thiagodelgado111** [comment]
Might be the same problem here: #25

**Comment by abernix** [comment]
@chmanie I changed [this line](https://github.com/DispatchMe/meteor-phantomjs-tests/blob/a659a87abf97fe433059fcadf54139032e4cc534/server.js#L18) to include the additional parameters: `--remote-debugger-port=9000 --remote-debugger-autorun=yes` (as separate array elements).  After that I was able to (kludgedly) use the [PhantomJS troubleshooting](http://phantomjs.org/troubleshooting.html) process to work through it.

I was was getting [this error](https://github.com/ariya/phantomjs/issues/13165) and ultimately decided that my `modules.js` was large(??) enough that it was getting corrupted due to an error in Qt(??) (which PhantomJS uses).   I also found that PhantomJS 2.0 wasn't had a hard time waiting until dependencies were loaded before it said it was ready.  As you can probably infer, I have some things that still aren't clear to me.  I ended up [switching this packages dependencies to PhantomJS 1.9.8](https://github.com/abernix/meteor-phantomjs-tests/commit/ced17025f338653a8fb7d31a280825eecd07e2ea) which solved the problem entirely for me.

**Comment by thiagodelgado111** [comment]
@abernix I tried to do that but when I try to open the debugging console at localhost:9000 it doesn't work :/

```

  const SERVER = '--host 127.0.0.1';
  const PORT = '--port 9000';
  const REMOTE_DEBUGGER_PORT = '--remote-debugger-port=9000';
  const REMOTE_DEBUGGER_AUTORUN = '--remote-debugger-autorun=yes';
  const phantomjsArgs = [scriptPath, SERVER, PORT, REMOTE_DEBUGGER_PORT, REMOTE_DEBUGGER_AUTORUN];
  console.log('args: ', phantomjsArgs);

  const phantomProcess = childProcess.execFile(phantomjs.path, phantomjsArgs, {
    env: {
      ROOT_URL: process.env.ROOT_URL,
    }
  });
```

**Comment by abernix** [comment]
Why are you adding `--host` and `--port`?  I'm not even sure those are [valid `phantomjs` flags](http://phantomjs.org/api/command-line.html), so it's possible the command is error-ing out completely.

I changed the line to this:

```js
const phantomProcess = childProcess.execFile(phantomjs.path, ['--remote-debugger-port=9000', '--remote-debugger-autorun=yes', scriptPath], {
```

...Ran it and `localhost:9000` worked fine.

**Comment by thiagodelgado111** [comment]
oops, sorry for that @abernix, moving `scriptPath` to the last arg position worked

**Comment by gferreri** [comment]
This error just popped up for me after adding `meteorhacks:kadira` to my project. I've tried enabling phantomJs debugging but nothing pops out to me. Interestingly, if I limit the number of server tests that run (using `it.only()`), then the client tests also run. But if I allow all 334 server tests to run, the client tests will not.

**Comment by ubald** [comment]
@gferreri I can confirm that on my end removing `meteorhacks:kadira` made the client tests run

**Comment by gferreri** [comment]
It appears to be some issue when running the client and server tests in parallel. I [forked](https://github.com/gferreri/meteor-mocha-phantomjs) the project and made it synchronous, so the client tests run before the server tests. Everything works now.

IMO it seems dangerous to run the client and server tests in parallel anyway, especially if the client tests are interacting with the server in some way. I suppose testing purists will tell you that your tests aren't designed properly if that's a problem for you.

**Comment by aldeed** [comment]
@gferreri We could merge a PR adding a command option for series vs. parallel server/client tests. I was on the fence about making them parallel initially, but it's faster and does not cause problems if you write tests carefully, so I think it should be the default with an option to switch to series.

**Comment by MuyBien** [comment]
Hi!
I have the same problem. I was hoping removing kadira will fix the problem like others but I still have the 2 client fail without any client test written... :(

What can I check now? Or is this another way to launch tests?

Thanks.
